File: app/src/main/java/com/aar/app/wordsearch/features/mainmenu/MainMenuViewModel.java
package com.aar.app.wordsearch.features.mainmenu;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.aar.app.wordsearch.data.GameThemeRepository;
import com.aar.app.wordsearch.model.GameTheme;

import java.util.List;

public class MainMenuViewModel extends ViewModel {

    private GameThemeRepository mGameThemeRepository;

    private MutableLiveData<List<GameTheme>> mOnGameThemeLoaded;

    public MainMenuViewModel(GameThemeRepository gameThemeRepository) {
        mGameThemeRepository = gameThemeRepository;
        mOnGameThemeLoaded = new MutableLiveData<>();
    }

    public void loadData() {
        mOnGameThemeLoaded.setValue(mGameThemeRepository.getGameThemes());
    }

    public LiveData<List<GameTheme>> getOnGameThemeLoaded() {
        return mOnGameThemeLoaded;
    }
}

File: app/src/main/java/co/africanwolf/hiddenwords/mainmenu/MainMenuViewModel.kt
package co.africanwolf.hiddenwords.mainmenu

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import co.africanwolf.hiddenwords.data.GameThemeRepository
import co.africanwolf.hiddenwords.model.GameTheme
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class MainMenuViewModel @Inject constructor(private val mGameThemeRepository: GameThemeRepository) : ViewModel() {

    private val mOnGameThemeLoaded: MutableLiveData<List<GameTheme>> = MutableLiveData()

    fun loadData() {
        mOnGameThemeLoaded.value = mGameThemeRepository.gameThemes
    }

    val onGameThemeLoaded: LiveData<List<GameTheme>>
        get() = mOnGameThemeLoaded
}
